src/lib.rs

```diff
@@ -465,7 +478,11 @@ pub enum Binding {
     /// Built-in shader variable.
     BuiltIn(BuiltIn),
     /// Indexed location.
-    Location(u32, Option<Interpolation>),
+    Location {
+        location: u32,
+        interpolation: Option<Interpolation>,
+        sampling: Option<Sampling>
```

Comment:
if I understand correctly, this has to be specified if `interpolation` is either `Some(Perspective)` or `Some(Linear)`, correct?

Reply:
Yes. An earlier version of this PR had the `Perspective` and `Linear` variants of `Interpolation` carry the `Sampling` values, but it was pretty unwieldy for the front ends. This code would be one of the sites to do the default resolution you referred to earlier, correct?

Reply:
Yes. I was just thinking if it makes sense to have `interpolation: Option<(Interpolation, Sampling)>`, and the downside would be that the frontends would have to specify the `Sampling` for `Flat`, which wouldn't be nice.
So I think what you have here is better.